Don't use the vendored tomli package - use our own package instead.

This patch exists only for automatic dependency detection. The vendored package
is dropped via the generate file, and the flit_core import code is written such
that when vendored tomli is not available, it uses the system one, so there is
no need for further changes.

--- flit_core-3.9.0/pyproject.toml
+++ flit_core-3.9.0/pyproject.toml
@@ -9,7 +9,9 @@ authors=[
     {name = "Thomas Kluyver & contributors", email = "thomas@kluyver.me.uk"},
 ]
 description = "Distribution-building parts of Flit. See flit package for more information"
-dependencies = []
+dependencies = [
+    'tomli',
+]
 requires-python = '>=3.6'
 readme = "README.rst"
 license = {file = "LICENSE"}
